### ISSUE TITLE
Python syntax warning on some az command groups

### ISSUE THREAD

**Issue by SaschaSchlemmer** [Azure/azure-cli#12035]
For me some command groups such as e.g. _webapp_ and _appservice_ (not an exhaustive list) any command causes python syntax warnings to appear. Example: 

```
az webapp --help
/usr/local/Cellar/azure-cli/2.0.80/libexec/lib/python3.8/site-packages/azure/storage/blob/_upload_chunking.py:403: SyntaxWarning: "is" with a literal. Did you mean "=="?
  if n is 0 or self._buffer.closed:
/usr/local/Cellar/azure-cli/2.0.80/libexec/lib/python3.8/site-packages/azure/storage/blob/baseblobservice.py:1009: SyntaxWarning: "is not" with a literal. Did you mean "!="?
  if lease_duration is not -1 and \
/usr/local/Cellar/azure-cli/2.0.80/libexec/lib/python3.8/site-packages/azure/storage/blob/baseblobservice.py:2660: SyntaxWarning: "is not" with a literal. Did you mean "!="?
  if lease_duration is not -1 and \
/usr/local/Cellar/azure-cli/2.0.80/libexec/lib/python3.8/site-packages/azure/storage/common/_connection.py:82: SyntaxWarning: "is" with a literal. Did you mean "=="?
  self.protocol = self.protocol if parsed_url.scheme is '' else parsed_url.scheme

Group
    az webapp : Manage web apps.

Subgroups:
    auth                        : Manage webapp authentication and authorization.
...
```

**Environment summary**
mac os Catalina (10.15.3)

**Comment by arrownj** [comment]
@SaschaSchlemmer , thanks for reporting, we are looking at it.

**Comment by TsuyoshiUshio** [comment]
I had the same issue on an Azure Pipeline Mac hosted agent. It seems the same version.

```
2020-02-07T18:25:59.5809080Z ##[error]Error: /usr/local/Cellar/azure-cli/2.0.80/libexec/lib/python3.8/site-packages/azure/multiapi/storage/v2018_11_09/blob/_upload_chunking.py:403: SyntaxWarning: "is" with a literal. Did you mean "=="?
  if n is 0 or self._buffer.closed:
/usr/local/Cellar/azure-cli/2.0.80/libexec/lib/python3.8/site-packages/azure/multiapi/storage/v2018_11_09/blob/baseblobservice.py:1063: SyntaxWarning: "is not" with a literal. Did you mean "!="?
  if lease_duration is not -1 and \
/usr/local/Cellar/azure-cli/2.0.80/libexec/lib/python3.8/site-packages/azure/multiapi/storage/v2018_11_09/blob/baseblobservice.py:2719: SyntaxWarning: "is not" with a literal. Did you mean "!="?
  if lease_duration is not -1 and \
/usr/local/Cellar/azure-cli/2.0.80/libexec/lib/python3.8/site-packages/azure/multiapi/storage/v2018_11_09/common/_connection.py:82: SyntaxWarning: "is" with a literal. Did you mean "=="?
  self.protocol = self.protocol if parsed_url.scheme is '' else parsed_url.scheme
/usr/local/Cellar/azure-cli/2.0.80/libexec/lib/python3.8/site-packages/azure/multiapi/cosmosdb/v2017_04_17/common/_connection.py:91: SyntaxWarning: "is" with a literal. Did you mean "=="?
  self.protocol = self.protocol if parsed_url.scheme is '' else parsed_url.scheme
```

**Comment by celaus** [comment]
Similarly on Linux:
~~~
$ az account list -o table
/usr/lib64/az/lib/python3.6/site-packages/jmespath/visitor.py:32: SyntaxWarning: "is" with a literal. Did you mean "=="?
  if x is 0 or x is 1:
/usr/lib64/az/lib/python3.6/site-packages/jmespath/visitor.py:32: SyntaxWarning: "is" with a literal. Did you mean "=="?
  if x is 0 or x is 1:
/usr/lib64/az/lib/python3.6/site-packages/jmespath/visitor.py:34: SyntaxWarning: "is" with a literal. Did you mean "=="?
  elif y is 0 or y is 1:
/usr/lib64/az/lib/python3.6/site-packages/jmespath/visitor.py:34: SyntaxWarning: "is" with a literal. Did you mean "=="?
  elif y is 0 or y is 1:
/usr/lib64/az/lib/python3.6/site-packages/jmespath/visitor.py:260: SyntaxWarning: "is" with a literal. Did you mean "=="?
  if original_result is 0:
~~~

I am not sure why it even uses different Python versions:

~~~
$ az --version
azure-cli                         2.0.80 *

command-modules-nspkg              2.0.3
core                              2.0.80 *
nspkg                              3.0.4
telemetry                          1.0.4

Extensions:
azure-cli-iot-ext                  0.5.2

Python location '/home/cm/.pyenv/versions/3.8.1/bin/python3'
Extensions directory '/home/cm/.azure/cliextensions'

Python (Linux) 3.8.1 (default, Jan 30 2020, 16:51:40) 
[GCC 9.2.1 20190827 (Red Hat 9.2.1-1)]

Legal docs and information: aka.ms/AzureCliLegal



You have 2 updates available. Consider updating your CLI installation. Instructions can be found at https://docs.microsoft.com/en-us/cli/azure/install-azure-cli

Please let us know how we are doing: https://aka.ms/clihats
~~~

**Comment by jgall** [comment]
I also have this issue on MacOS 10.15.4 with the latest Homebrew install of azure-cli.

**Comment by JerryJoyce** [comment]
I hit this today.  Fresh install of Catalina using azure cli from Homebrew

**Comment by sschlesier** [comment]
I now have this as well.  I updated to Fedora 32 and with it came Python 3.8 and these SyntaxWarnings.

I tried updating to the latest Azure CLI 2.6.0 but no change for my case.

I'm getting

```
/usr/lib64/az/lib/python3.6/site-packages/jmespath/visitor.py:32: SyntaxWarning: "is" with a literal. Did you mean "=="?
/usr/lib64/az/lib/python3.6/site-packages/jmespath/visitor.py:34: SyntaxWarning: "is" with a literal. Did you mean "=="?
  elif y is 0 or y is 1:
/usr/lib64/az/lib/python3.6/site-packages/jmespath/visitor.py:34: SyntaxWarning: "is" with a literal. Did you mean "=="?
  elif y is 0 or y is 1:
/usr/lib64/az/lib/python3.6/site-packages/jmespath/visitor.py:260: SyntaxWarning: "is" with a literal. Did you mean "=="?
  if original_result is 0:
/usr/lib64/az/lib/python3.6/site-packages/jmespath/visitor.py:32: SyntaxWarning: "is" with a literal. Did you mean "=="?
  if x is 0 or x is 1:
/usr/lib64/az/lib/python3.6/site-packages/jmespath/visitor.py:32: SyntaxWarning: "is" with a literal. Did you mean "=="?
  if x is 0 or x is 1:
/usr/lib64/az/lib/python3.6/site-packages/jmespath/visitor.py:34: SyntaxWarning: "is" with a literal. Did you mean "=="?
  elif y is 0 or y is 1:
/usr/lib64/az/lib/python3.6/site-packages/jmespath/visitor.py:34: SyntaxWarning: "is" with a literal. Did you mean "=="?
  elif y is 0 or y is 1:
/usr/lib64/az/lib/python3.6/site-packages/jmespath/visitor.py:260: SyntaxWarning: "is" with a literal. Did you mean "=="?
  if original_result is 0:
```

which is generated by several calls to 

`az servicebus topic subscription list ...`

**Comment by cajund** [comment]
If the warnings are getting in the way, you can suppress them:

`export PYTHONWARNINGS="ignore"`

**Comment by one-random-dev** [comment]
also hit this today, while performing an upload-batch to a storage account fileshare

`az storage file upload-batch --account-name managementops --destination destionation --source source  --pattern '*.app'
/usr/local/Cellar/azure-cli/2.8.0/libexec/lib/python3.8/site-packages/azure/multiapi/storage/v2018_11_09/blob/_upload_chunking.py:403: SyntaxWarning: "is" with a literal. Did you mean "=="?
  if n is 0 or self._buffer.closed:
/usr/local/Cellar/azure-cli/2.8.0/libexec/lib/python3.8/site-packages/azure/multiapi/storage/v2018_11_09/blob/baseblobservice.py:1063: SyntaxWarning: "is not" with a literal. Did you mean "!="?
  if lease_duration is not -1 and \
/usr/local/Cellar/azure-cli/2.8.0/libexec/lib/python3.8/site-packages/azure/multiapi/storage/v2018_11_09/blob/baseblobservice.py:2719: SyntaxWarning: "is not" with a literal. Did you mean "!="?
  if lease_duration is not -1 and \
/usr/local/Cellar/azure-cli/2.8.0/libexec/lib/python3.8/site-packages/azure/multiapi/storage/v2018_11_09/common/_connection.py:82: SyntaxWarning: "is" with a literal. Did you mean "=="?
  self.protocol = self.protocol if parsed_url.scheme is '' else parsed_url.scheme
/usr/local/Cellar/azure-cli/2.8.0/libexec/lib/python3.8/site-packages/azure/multiapi/cosmosdb/v2017_04_17/common/_connection.py:91: SyntaxWarning: "is" with a literal. Did you mean "=="?
  self.protocol = self.protocol if parsed_url.scheme is '' else parsed_url.scheme
There is no credential provided in your command and environment, we will query account key for your storage account. 
Please provide --connection-string, --account-key or --sas-token as credential, or use `--auth-mode login` if you have required RBAC roles in your command. For more information about RBAC roles in stoarge, you can see https://docs.microsoft.com/en-us/azure/storage/common/storage-auth-aad-rbac-cli. 
Setting corresponding environment variable can avoid inputting credential in your command. Please use --help to get more information.
[]`

the issue is that the upload did not go trough

**Comment by panchagnula** [comment]
This is not app service but package issue I recommend using version 3.7 of Python on your Mac instead of 3.8.*

**Comment by fengzhou-msft** [comment]
`azure-storage-blob` is replaced by `azure-multiapi-storage` and the syntax warning will be fixed by https://github.com/Azure/azure-multiapi-storage-python/pull/31

**Comment by yungezz** [comment]
hi @fengzhou-msft https://github.com/Azure/azure-multiapi-storage-python/pull/31 is merged. Can this issue been closed now? thanks.

**Comment by fengzhou-msft** [comment]
Resolved by #14785, please try with Azure CLI 2.11.0.
